### PR TITLE
change default log date format, and write log to file

### PR DESCRIPTION
**Pull request by brownchow**
Hi all:
1. I modified the default date format, to make it more human readable 

date format befor:
```sh
{"level":"info","ts":1606734528.80338,"caller":"gilfoyle/serve.go:32","msg":"Initializing API server"}
{"level":"info","ts":1606734528.80338,"caller":"gilfoyle/serve.go:33","msg":"Environment","debug":false}
```
date format now:
```sh
{"level":"info","time":"2020-11-30T19:12:56.139+0800","caller":"gilfoyle/serve.go:32","message":"Initializing API server"}
{"level":"info","time":"2020-11-30T19:12:56.141+0800","caller":"gilfoyle/serve.go:33","message":"Environment","debug":false}
```

2. use  log rolling package  `gopkg.in/natefinch/lumberjack` write log to file, default is {ProjectRootDir}/logs/server.log

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -21,8 +33,43 @@ func NewLogger() (*zap.Logger, error) {
...
 	loggerOnce.Do(func() {
-		Logger, err = zap.NewProduction()
+		lumberJackLogger := lumberjack.Logger{
+			Filename:   "logs/server.log", //Filename is the file to write logs to
+			MaxSize:    50,                //MB
+			MaxBackups: 30,                //MaxBackups is the maximum number of old log files to retain.
+			MaxAge:     90,                //days
+			Compress:   false,
+		}
+		zap.RegisterSink("lumberjack", func(*url.URL) (zap.Sink, error) {
```
Comment (sundowndev) [inline logger.go]: Error check is missing, could you please assign top-level err variable here then return?

File: logger.go [hunk 2]
```diff
@@ -21,8 +33,43 @@ func NewLogger() (*zap.Logger, error) {
...
+			return lumberjackSink{
+				Logger: &lumberJackLogger,
+			}, nil
+		})
+		zapConfig := zap.Config{
+			Level:             zap.NewAtomicLevelAt(zapcore.DebugLevel),
+			Development:       false,
```
Comment (sundowndev) [inline logger.go:63]: This value should be set according to Config.Settings.Debug (https://github.com/dreamvo/gilfoyle/blob/master/config/config.go#L21) value

**Comment by sundowndev** [comment]
Hi, thanks for this awesome first contribution!

I posted few comments, lint errors need to be fixed before merging, you may also use the global config to set some config values here. I'll try to merge this before #36.

File: logger.go (hunk 1 above)
Comment (brownchow) [reply logger.go]: OK, I will refactor code and check zap.RegisterSink( ) returned err

File: logger.go (hunk 2 above)
Comment (brownchow) [reply logger.go:63]: I will set the LogLevel from the config file

**Comment by sundowndev** [comment]
LGTM :confetti_ball:

**Review by sundowndev — APPROVED** [review]

**Pull request merged**